driver-sync/src/main/com/mongodb/client/internal/ClientEncryptionImpl.java

```diff
@@ -171,6 +183,65 @@ public RewrapManyDataKeyResult rewrapManyDataKey(final Bson filter, final Rewrap
         return new RewrapManyDataKeyResult(collection.bulkWrite(updateModels));
     }
 
+    @Override
+    public BsonDocument createEncryptedCollection(final MongoDatabase database, final String collectionName,
+            final CreateCollectionOptions createCollectionOptions, final CreateEncryptedCollectionParams createEncryptedCollectionParams)
+            throws MongoUpdatedEncryptedFieldsException {
+        notNull("collectionName", collectionName);
+        notNull("createCollectionOptions", createCollectionOptions);
+        notNull("createEncryptedCollectionParams", createEncryptedCollectionParams);
+        MongoNamespace namespace = new MongoNamespace(database.getName(), collectionName);
+        Bson rawEncryptedFields = createCollectionOptions.getEncryptedFields();
+        if (rawEncryptedFields == null) {
+            throw new MongoConfigurationException(format("`encryptedFields` is not configured for the collection %s.", namespace));
+        }
+        CodecRegistry codecRegistry = options.getKeyVaultMongoClientSettings() == null
+                ? MongoClientSettings.getDefaultCodecRegistry()
+                : options.getKeyVaultMongoClientSettings().getCodecRegistry();
+        BsonDocument actualEncryptedFields = rawEncryptedFields.toBsonDocument(BsonDocument.class, codecRegistry).clone();
+        BsonValue fields = actualEncryptedFields.get("fields");
+        if (fields != null) {
+            if (!fields.isArray()) {
+                throw new MongoConfigurationException(format("`encryptedFields` is incorrectly configured for the collection %s."
+                        + " `encryptedFields.fields` must be an array, but is %s type.", namespace, fields.getBsonType()));
```

Comment:
This exception is not according to the current spec, and comes from the following:

- https://mongodb.slack.com/archives/C0406ECL478/p1675269046945409
- https://jira.mongodb.org/browse/DRIVERS-2540 (does not have a completed spec change), which proposes to throw in the `createCollection` methods. And I [think](https://jira.mongodb.org/browse/DRIVERS-2540?focusedCommentId=5159364&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-5159364) that if `createCollection` methods need to throw, so should the `createEncryptedCollection` method.

If there is a concern that we should not throw this exception until (if ever) it is approved as a spec change, I will undo.

Reply:
I undone this part because I [realized](https://jira.mongodb.org/browse/DRIVERS-2540?focusedCommentId=5180748&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-5180748) that the proposal in DRIVERS-2540 is not fully-baked.